src/layer/BYOCLayer.ts

```diff
@@ -81,6 +81,7 @@ export class BYOCLayer extends AbstractSentinelHubV3Layer {
       if (this.collectionId === null || this.evalscript === null || this.subType === null) {
         const layerParams = await this.fetchLayerParamsFromSHServiceV3(innerReqConfig);
         this.collectionId = layerParams['collectionId'];
+        this.legend = layerParams['legend'] ? layerParams['legend'] : null;
```

Comment:
I would follow the pattern and check if value for `legend` has already been set on layer.  Not sure if really needed, fftr 
```javascript
if (!this.legend){
  this.legend = layerParams['legend'] ? layerParams['legend'] : null;
}
```

Reply:
Done that.